src/change_stream.ts

```diff
@@ -304,14 +295,86 @@ export class ChangeStream extends EventEmitter {
       const cursor = this.cursor;
 
       return cursor.close(err => {
-        ['data', 'close', 'end', 'error'].forEach(event => cursor.removeAllListeners(event));
+        this[kStream]?.cleanupListeners();
         this.cursor = undefined;
 
         return cb(err);
       });
     });
   }
 
+  /**
+   * Return a modified Readable stream including a possible transform method.
+   * @throws MongoError if this.cursor is undefined
+   */
+  stream(options?: StreamOptions): ChangeStreamStream {
```

Comment:
while we're here, do you think we should rename `StreamOptions` to something less generic, maybe `CursorStreamOptions`?